### DISCUSSION TITLE
EDA Artifacts

### DISCUSSION THREAD

**Discussion by BaggioMarco** [neuropsychology/NeuroKit#523] [Q&A] [answered]
Hi,

Is your package capable to deal with the following EDA artifacts? 
E.g. "repairing artifacts" when is possibile or "cleaning/remove" when is not possible repair artifacts.

- High-frequency noise
- Rapid-transient artifact
- Loose electrodes, decoupling from skin
- Electrode movement

Source: https://www.tobiipro.com/learn-and-support/learn/GSR-essentials/gsr-artifacts/

thank you in advance

**Comment by DominiqueMakowski** [comment, marked as answer]
Most of these artifacts are "repaired" through filtering, aside from the bad periods of signal related to decoupling or disconnection (this you would have to do manually, we don't have currently functions to detect and fix bad periods of signal - though that'd be something that we could add (maybe we should open an issue for that @zen-juen @Tam-Pham )). 

Note that repairing artifacts is not a silver bullet, bad/artifacted signal will always limit your ability to get true results (and can even lead to some nonsensical results). As they say in the machine learning world, "garbage in, garbage out" 😉 



Also note to ourselves @zen-juen @Tam-Pham : https://www.tobiipro.com/learn-and-support/learn/GSR-essentials/gsr-artifacts/ is a nice article with clear illustrations. We might want to 1) reference it somewhere in our tutorial and 2) "borrow" (?) or reproduce (probably safer) some of their figures (we might have examples of such artifacts in our data).
